samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+|                    | EdDSA       | `RFC 8032 - Edwards-Curve Digital Signature Algorithm (EdDSA)`_            | `RFC 8032 - Edwards-Curve Digital Signature Algorithm (EdDSA)`_            |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+| Hash               | SHA256      | `NIST - SHA`_                                                              | `NIST - SHA`_                                                              |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | SHA512      |                                                                            | `NIST - SHA`_                                                              |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+| HMAC               | HMAC SHA256 | `NIST - HMAC`_                                                             | `NIST - HMAC`_                                                             |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HMAC SHA512 |                                                                            | `NIST - HMAC`_                                                             |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+| HKDF               | HKDF SHA256 | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+|                    | HKDF SHA512 |                                                                            | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+
+
+Requirements
+************
+
+* One of the following development boards:
+
+  * |nRF9160DK|
+  * |nRF5340DK|
+  * |nRF52840DK|
+  * |nRF52DK|
+  * |nRF51DK|
+
+* something else??
+
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/bluetooth/central_bas`
```

Comment:
I am sure there is some standard page in Zephyr or nRF Connect SDK listing how to build a generic application

The input that we need to address is the options you can build the crypto test application through their overlay-files

The options are:

- overlay-backend-sw.conf - SW only (except a hw-module to generate entropy, read: rng)
- overlay-backend-hw.conf - HW accelerated through the use of Arm CryptoCell accelerator (for cryptography and entropy)
- overlay-backend-hw-sw.conf - A combination of HW acceleration through Arm CryptoCell and SW implementation (which adds key-sizes and/or algorithms not supported in CryptoCell).

The special case overlay-backend-hw-sw.conf should give a link to the glue-mechanism inside nrf_security for reference

-

Reply:
link to nrfxlib: nRF Security -> Configuration

Reply:
overlay files -> the 'Building and programming a sample application' page might need an update?

Reply:
link to 'Building and programming a sample application' and add a list of overlay files ,like in https://docs.zephyrproject.org/latest/samples/net/sockets/echo_client/README.html

Reply:
done